src/test/java/org/fcrepo/migration/PicocliIT.java

```diff
@@ -113,4 +127,124 @@ public void testExistingRepoDifferentStorageLayout() throws Exception {
+        assertEquals(1, result);
+    }
+
+    @Test
+    public void testSha256DigestAlgorithm() throws Exception {
+        final Path targetDir = tmpDir.resolve("target");
+        final Path workingDir = tmpDir.resolve("working");
+        final String[] args = {"--target-dir", targetDir.toString(), "--working-dir", workingDir.toString(),
+                "--source-type", "LEGACY","--migration-type", "PLAIN_OCFL",
+                "--datastreams-dir","src/test/resources/legacyFS/datastreams/2015/0430/16/01",
+                "--objects-dir", "src/test/resources/legacyFS/objects/2015/0430/16/01",
+                "--algorithm", "sha256"};
+        final PicocliMigrator migrator = new PicocliMigrator();
+        final CommandLine cmd = new CommandLine(migrator);
+
+        final int result = cmd.execute(args);
+        assertEquals(0, result);
+        assertTrue(Files.list(targetDir).anyMatch(element -> element.endsWith("0=ocfl_1.0")));
+        final Path baseDir = targetDir.resolve("5b5").resolve("62d").resolve("d69")
+                .resolve("5b562dd698f17e3198e007e6f77f9e48f20a556c6bae84e6fc8d98544831daa6");
+        final File inventory = baseDir.resolve("inventory.json").toFile();
+        assertTrue(inventory.exists());
+        validateManifests(inventory, "SHA-256", baseDir);
+    }
+
+    /**
+     * Validate the manifest digests in an inventory file.
+     * @param inventory the inventory file
+     * @param digestAlgo the digest algorithm
+     * @param baseDir the path of the OCFL object
+     * @throws IOException issues opening the inventory file.
+     * @throws NoSuchAlgorithmException issues creating a MessageDigest.
+     */
+    private void validateManifests(final File inventory, final String digestAlgo, final Path baseDir)
+            throws IOException, NoSuchAlgorithmException {
+        final var manifests = getManifests(inventory);
+        final MessageDigest md = MessageDigest.getInstance(digestAlgo);
+        for (final var entry : manifests.entrySet()) {
+            final File f = baseDir.resolve(entry.getKey()).toFile();
+            assertTrue(f.exists());
+            final String digest = getDigest(new FileInputStream(f), md, 2048);
+            assertEquals(entry.getValue(), digest);
+        }
+    }
+
+    /**
+     * Utility to get a message digest for a file.
+     * @param stream the input stream of the file
+     * @param md the message digest to calculate
+     * @param byteSize size of file to read per loop.
+     * @return the hash
+     * @throws IOException problems opening the inputstream
+     */
+    private String getDigest(final InputStream stream, final MessageDigest md, final int byteSize) throws IOException {
+        md.reset();
+        final byte[] byteArray = new byte[byteSize];
+        int bytes;
+        while ((bytes = stream.read(byteArray)) != -1) {
+            md.update(byteArray, 0, bytes);
+        }
+        return new String(Hex.encodeHex(md.digest()));
+    }
```

Comment:
I think you could replace this method with `Hex.encodeDex(DigestUtils.digest(md, stream))`